scipy/optimize/slsqp.py

```diff
@@ -440,7 +474,9 @@ def cjac(x, *args):
         if majiter > majiter_prev:
             # call callback if major iteration has incremented
             if callback is not None:
-                callback(np.copy(x))
+                new_x = np.copy(full_x0)
+                new_x[~filtered_params] = x
+                callback(new_x)
```

Comment:
Why not use `_filter_params_func` on `callback` when it is passed in?

Reply:
Great idea.